yarn-project/circuits.js/src/abis/abis.ts

```diff
@@ -94,13 +94,10 @@ export async function computeFunctionLeaf(wasm: CircuitsWasm, fnLeaf: FunctionLe
 }
 
 export async function computeFunctionTreeRoot(wasm: CircuitsWasm, fnLeafs: Fr[]) {
-  const inputBuf = serializeToBuffer(fnLeafs);
+  const inputVector = serializeBufferArrayToVector(fnLeafs.map(fr => fr.toBuffer()));
   wasm.call('pedersen__init');
-  const outputBuf = wasm.call('bbmalloc', 32);
-  const inputBufPtr = wasm.call('bbmalloc', inputBuf.length);
-  wasm.writeMemory(inputBufPtr, inputBuf);
-  await wasm.asyncCall('abis__compute_function_tree_root', inputBufPtr, fnLeafs.length, outputBuf);
-  return Fr.fromBuffer(Buffer.from(wasm.getMemorySlice(outputBuf, outputBuf + 32)));
+  const result = await wasmAsyncCall(wasm, 'abis__compute_function_tree_root', { toBuffer: () => inputVector }, 32);
```

Comment:
I feel this `{ toBuffer: () => val }` hack could be replaced with this just accepting the buffer in the first place
```
export async function wasmAsyncCall(
  wasm: AsyncWasmWrapper,
  fnName: string,
  input: { toBuffer: () => Buffer },
  expectedOutputLength: number,
): Promise<Buffer> {
  const inputData = input.toBuffer();

```
The function seems to immediately take the buffer in the first place?

Reply:
Yeah I think its saying "input object should have a toBuffer function". I just pushed a change to have the wasm call helpers accept `Buffer | { toBuffer: () => Buffer }` instead.

Tagging @PhilWindle since it looks like he may have been the original author of these (or at least he pulled them from AC).

Updated `wasmCall` and `wasmAsyncCall` in `abis.ts` along with some calls to those helpers. Also updated `wasmCall` in `rollup_wasm_wrapper.ts` ([example](https://github.com/AztecProtocol/aztec3-packages/pull/314/files#diff-b850c43ea46f903b65fe13a4593c8611cd029a0e45db760e50cc109abad39c08R18))

Reply:
Phil gave this the thumbs up